impl/src/main/java/org/jboss/weld/bean/proxy/InterceptedSubclassFactory.java

```diff
@@ -240,8 +240,15 @@ void doReturn(CodeAttribute b, ClassMethod method) {
     private boolean bridgeMethodsContainsMethod(Set<BridgeMethod> processedBridgeMethods, MethodSignature signature, Type returnType) {
         for (BridgeMethod bridgeMethod : processedBridgeMethods) {
             if (bridgeMethod.signature.equals(signature)) {
+                // method signature is equal (name and params) but return type can still differ
                 if (returnType != null) {
-                    return bridgeMethod.returnType.equals(returnType) ? true : false;
+                    if (bridgeMethod.returnType.equals(Object.class)) {
+                        // bridge method with matching signature has parameterized type as return type, this is a match
```

Comment:
I think the return type is not a parameterized type but a type variable.

Reply:
Yea, that's what I meant. Thanks. :)